outputs/N2/parameters.txt
###    MODEL PARAMETERS
ND                    =   101
NLOOPS                =   16
NMINSTEPS             =   500
NMAXSTEPS             =   500
TCONV                 =   0.01
Resuming from previous?   False

### PLANET/ATMOSPHERE PARAMETERS
Radius [km]           =   6371.0
G                     =   980
Separation (AU)       =   1.0
Surface Albedo        =   0.23
Instellation [Solar]  =   1.0
TOA Pressure [atm]    =   1e-05
Surface Pressure[atm] =   1.0
TOA Temperature [K]   =   200
Surface Temp [K]      =   280
Tropopause layer      =   22
Argon Mixing Ratio    =   0.02
Ozone Flag            =   False
Methane Flag          =   True
Surface rel.humidity  =   0.7
Energy cons. flag     =   True
Rainout Flag          =   True
Dry Adiabat Flag      =   False
Fractional cloudiness =   0.0

###    EDDYSED PARAMETERS
eddysed flag          =   true
Fracional cloudiness  =   0.0
Min. eddy diffusivity =   100000.0
CRAINF                =   0.01
CSIG                  =   2.0
SUPERSAT              =   0.0
Min cold trap fH2O    =   4e-10
Min upper atm. FC frac=   0.01

###    FILE PATHS
Python File Path      =   /Users/cooper/Documents/CLIMA-MEAC-coupling
Cloudy CLIMA Path     =   /Users/cooper/Documents/CLIMA-MEAC-coupling/cloudy_clima
MEAC Path             =   /Users/cooper/Documents/CLIMA-MEAC-coupling/hu-code-sr

###    CLIMA FILE PATHS
CLIMA IO Folder       =   /Users/cooper/Documents/CLIMA-MEAC-coupling/cloudy_clima/CLIMA/IO
CLIMA header file     =   /Users/cooper/Documents/CLIMA-MEAC-coupling/cloudy_clima/CLIMA/INCLUDE/header.inc
CLIMA Input File      =   /Users/cooper/Documents/CLIMA-MEAC-coupling/cloudy_clima/CLIMA/IO/input_clima.dat
CLIMA Mixing Ratios   =   /Users/cooper/Documents/CLIMA-MEAC-coupling/cloudy_clima/CLIMA/IO/mixing_ratios.dat
CLIMA Last Step Output=   /Users/cooper/Documents/CLIMA-MEAC-coupling/cloudy_clima/CLIMA/IO/clima_last.tab
CLIMA Allout File     =   /Users/cooper/Documents/CLIMA-MEAC-coupling/cloudy_clima/CLIMA/IO/clima_allout.tab
CLIMA TempIn File     =   /Users/cooper/Documents/CLIMA-MEAC-coupling/cloudy_clima/CLIMA/IO/TempIn.dat
CLIMA Tempout File    =   /Users/cooper/Documents/CLIMA-MEAC-coupling/cloudy_clima/CLIMA/IO/TempOut.dat
CLIMA Profiles Folder =   /Users/cooper/Documents/CLIMA-MEAC-coupling/cloudy_clima/CLIMA/IO/Profiles

###    MEAC FILE PATHS
MEAC Scenario Name    =   Sun/N2-Full
MEAC Scenario Path    =   scenario_library/Sun/N2-Full
MEAC Converged File   =   /Users/cooper/Documents/CLIMA-MEAC-coupling/hu-code-sr/scenario_library/Sun/N2-Full/ConcentrationSTD_base.dat
MEAC Scenario File    =   scenario_library/Sun/N2-Full/planet_scenario_N2r.h
MEAC Species File     =   scenario_library/Sun/N2-Full/species_scenario_N2r.dat
MEAC ConcentrationFile=   /Users/cooper/Documents/CLIMA-MEAC-coupling/hu-code-sr/scenario_library/Sun/N2-Full/ConcentrationSTD.dat
MEAC T-P Profile      =   scenario_library/Sun/N2-Full/TP.dat


outputs/CO2/parameters.txt
###    MODEL PARAMETERS
ND                    =   101
NLOOPS                =   16
NMINSTEPS             =   500
NMAXSTEPS             =   500
TCONV                 =   0.01
Resuming from previous?   False

### PLANET/ATMOSPHERE PARAMETERS
Radius [km]           =   6371.0
G                     =   980
Separation (AU)       =   1.0
Surface Albedo        =   0.23
Instellation [Solar]  =   1.0
TOA Pressure [atm]    =   1e-05
Surface Pressure[atm] =   1.0
TOA Temperature [K]   =   200
Surface Temp [K]      =   300
Tropopause layer      =   22
Argon Mixing Ratio    =   0.02
Ozone Flag            =   True
Methane Flag          =   True
Surface rel.humidity  =   0.7
Energy cons. flag     =   True
Rainout Flag          =   True
Dry Adiabat Flag      =   False
Fractional cloudiness =   0.0

###    EDDYSED PARAMETERS
eddysed flag          =   true
Fracional cloudiness  =   0.0
Min. eddy diffusivity =   100000.0
CRAINF                =   0.01
CSIG                  =   2.0
SUPERSAT              =   0.0
Min cold trap fH2O    =   4e-10
Min upper atm. FC frac=   0.01

###    FILE PATHS
Python File Path      =   /Users/cooper/Documents/CLIMA-MEAC-coupling
Cloudy CLIMA Path     =   /Users/cooper/Documents/CLIMA-MEAC-coupling/cloudy_clima
MEAC Path             =   /Users/cooper/Documents/CLIMA-MEAC-coupling/hu-code-sr

###    CLIMA FILE PATHS
CLIMA IO Folder       =   /Users/cooper/Documents/CLIMA-MEAC-coupling/cloudy_clima/CLIMA/IO
CLIMA header file     =   /Users/cooper/Documents/CLIMA-MEAC-coupling/cloudy_clima/CLIMA/INCLUDE/header.inc
CLIMA Input File      =   /Users/cooper/Documents/CLIMA-MEAC-coupling/cloudy_clima/CLIMA/IO/input_clima.dat
CLIMA Mixing Ratios   =   /Users/cooper/Documents/CLIMA-MEAC-coupling/cloudy_clima/CLIMA/IO/mixing_ratios.dat
CLIMA Last Step Output=   /Users/cooper/Documents/CLIMA-MEAC-coupling/cloudy_clima/CLIMA/IO/clima_last.tab
CLIMA Allout File     =   /Users/cooper/Documents/CLIMA-MEAC-coupling/cloudy_clima/CLIMA/IO/clima_allout.tab
CLIMA TempIn File     =   /Users/cooper/Documents/CLIMA-MEAC-coupling/cloudy_clima/CLIMA/IO/TempIn.dat
CLIMA Tempout File    =   /Users/cooper/Documents/CLIMA-MEAC-coupling/cloudy_clima/CLIMA/IO/TempOut.dat
CLIMA Profiles Folder =   /Users/cooper/Documents/CLIMA-MEAC-coupling/cloudy_clima/CLIMA/IO/Profiles

###    MEAC FILE PATHS
MEAC Scenario Name    =   Sun/CO2-Full
MEAC Scenario Path    =   scenario_library/Sun/CO2-Full
MEAC Converged File   =   /Users/cooper/Documents/CLIMA-MEAC-coupling/hu-code-sr/scenario_library/Sun/CO2-Full/ConcentrationSTD_base.dat
MEAC Scenario File    =   scenario_library/Sun/CO2-Full/planet_scenario_CO2r_emi.h
MEAC Species File     =   scenario_library/Sun/CO2-Full/species_scenario_CO2N2r_emi.dat
MEAC ConcentrationFile=   /Users/cooper/Documents/CLIMA-MEAC-coupling/hu-code-sr/scenario_library/Sun/CO2-Full/ConcentrationSTD.dat
MEAC T-P Profile      =   scenario_library/Sun/CO2-Full/TP.dat


parameters.py
import os

# Coupling parameters
ND              =   101                     # CLIMA layer variables DONT TOUCH
NLOOPS          =   16                       # Number of CLIMA-MEAC loops
NMINSTEPS       =   500                      # Minimum number of CLIMA steps per loop
NMAXSTEPS       =   500                      # Max Number of CLIMA steps per loops
NMAXT           =   1e50                    # Max MEAC run cumulative timestep
TCONV           =   1E-2                    # CLIMA convergence criterion, delta temperature

# Planet + atmosphere parameters
MASS            =   5.9722e24               # Planet mass in kg
RAD             =   6371.0                  # Planet radius in km
G               =   980                     # grav*M/R**2; Surface gravity, cgs
A               =   1.0                     # Semimajor axis in AU
SURFALB         =   0.23                     # Planet surface albedo
INSTELL         =   1.0                     # Planet instelation, relative to Earth
P0              =   1e-5                    # Top-of-atmosphere pressure [atm]
PSURF           =   1e+0                    # Surface pressure [atm]
T0              =   200                     # Top-of-atmosphere temperature [K]
TSURF           =   300                     # Surface temperature [K]
TROPOPAUSE      =   22                      # CLIMA tropopause layer, default 22 (of 101)
AR              =   2e-2                    # Argon mixing ratio
RELHUM          =   0.7                     # Surface relative humidity
FIXH2O          =   1                       # Fixed H2O flag. I really recommend leaving this on
IO3             =   1                       # Ozone flag -- IO3=0 means ozone isn't read in
IME             =   1                       # Methane/ethane flag

# Eddysed variables
DOEDDY          =   'true'                  # eddysed flag; if false, nothing below this matters
FCLOUD          =   0.0                     # fractional cloud coverage   
KZ_MIN          =   100000.0                # Minimum eddydiffusivity
CRAINF          =   0.01                    #
CSIG            =   2.0                     #
SUPERSAT        =   0.0                     #
COLDTRAPMINMIX  =   4e-10                   # Minimum coldtrap water mass mixing ratio
FCMINF          =   0.01                    # Minimum fraction for FC for upper atmosphere

# Misc
RESUMERUN       =   0                       # Continue from most recent run
ICONSERVE       =   1                       # CLIMA energy conservation flag, PLEASE KEEP ON
RAINOUT         =   1                       # controls rainout in 'main.c'. not used yet
ADIABATIC       =   0                       # tells CLIMA that atmosphere is dry adiabat  
ZMAX            =   50                      # Upper altitude limit, WIP
atm2Pa          =   101_325

#######################
###  Paths & Files  ###
#######################

# Paths
NAME            =   'CO2'
OUTPUT          =   "outputs/"+NAME
PATH            =   os.getcwd()                                     # Path to this python file
CLIMAPATH       =   f'{PATH}/cloudy_clima'                          # Path to the cloudy-CLIMA folder
MEACPATH        =   f'{PATH}/hu-code-sr'                            # Path to the MEAC folder

# CLIMA files
CINOUT          =   f'{CLIMAPATH}/CLIMA/IO'
CINCLUDE        =   f'{CLIMAPATH}/CLIMA/INCLUDE/header.inc'         # file with step count, conv. criterion
CINPUT          =   f'{CINOUT}/input_clima.dat'                     # input file
CMIXING         =   f'{CINOUT}/mixing_ratios.dat'                   # mixing ratios file
CLAST           =   f'{CINOUT}/clima_last.tab'                      # output file
CALLOUT         =   f'{CINOUT}/clima_allout.tab'                    # allout file
CTEMPIN         =   f'{CINOUT}/TempIn.dat'                          # temperature and water profile input
CTEMPOUT        =   f'{CINOUT}/TempOut.dat'                         # temperature and water profile output
C_O2            =   f'{CINOUT}/Profiles/O2.dat'                     # O2   vertical profile
C_N2            =   f'{CINOUT}/Profiles/N2.dat'                     # N2   vertical profile
C_H2            =   f'{CINOUT}/Profiles/H2.dat'                     # H2   vertical profile
C_CO2           =   f'{CINOUT}/Profiles/CO2.dat'                    # CO2  vertical profile
C_CH4           =   f'{CINOUT}/Profiles/CH4.dat'                    # CH4  vertical profile
C_C2H6          =   f'{CINOUT}/Profiles/C2H6.dat'                   # C2H6 vertical profile
C_H2O           =   f'{CINOUT}/Profiles/H2O.dat'                    # H2O  vertical profile
C_O3            =   f'{CINOUT}/Profiles/O3.dat'                     # O3   vertical profile

# MEAC files
MSCENARIONAME   =   "Sun/CO2-Full"                                   # Name of MEAC scenario folder

MSCENARIOPATH   =   f'scenario_library/{MSCENARIONAME}'             # Path to MEAC scenario folder
MZTP            =   f'{MSCENARIOPATH}/TP.dat'                       # MEAC ztp profile
for file in os.listdir(f"{MEACPATH}/{MSCENARIOPATH}"):              # Automatically identify key files
    if 'base' in file: MCONV = f"{MEACPATH}/{MSCENARIOPATH}/{file}"                 # Converged "seed" concentration file
    elif 'Concentration' in file: MCONC= f"{MEACPATH}/{MSCENARIOPATH}/{file}"       # Updated concentration file
    if 'planet' in file: MSCENARIO = f"{MSCENARIOPATH}/{file}"                      # Planet scenario file
    if 'species' in file: MSPECIES = f"{MSCENARIOPATH}/{file}"                      # Species file

# MCONV           =   f"{MEACPATH}/{MSCENARIOPATH}/ConcentrationSTD_base.dat"    # Conc. file from last convergence
# MSCENARIO       =   f'{MSCENARIOPATH}/planet_scenario_N2r.h'    # MEAC scenario file with planet parameters
# MSPECIES        =   f'{MSCENARIOPATH}/species_scenario_N2r.dat'       # MEAC atmosphere species file
# MCONC           =   f'{MEACPATH}/{MSCENARIOPATH}/ConcentrationSTD.dat'         # MEAC concentrations file
# MZTP            =   f'{MSCENARIOPATH}/TP.dat'                       # MEAC ztp profile